service/src/test/java/com/commercetools/pspadapter/tenant/TenantFactoryTest.java

```diff
@@ -131,14 +154,121 @@ public void createPaymentDispatcher() throws Exception {
+                verify(payonePostService, times(1)).executePost(payoneRequestCaptor.capture());
+                assertThat(payoneRequestCaptor.getValue()).isInstanceOf(AuthorizationRequest.class);
+                AuthorizationRequest request = (AuthorizationRequest) payoneRequestCaptor.getValue();
+                String expectedPayoneRequestType = mapCtpTransactionToPayoneRequestType(supportedTransactionType);
+                String actualPayoneRequestType = request.getRequest();
+                assertThat(actualPayoneRequestType)
+                        .withFailMessage(format("Unexpected Payone request type for respective CTP transaction [%s] in payment type [%s].\n\tExpected: [%s]\n\tActual: [%s]",
+                                supportedTransactionType.toSphereName(), supportedPaymentMethod.getKey(), expectedPayoneRequestType, actualPayoneRequestType))
+                        .isEqualTo(expectedPayoneRequestType);
+            }
+        }
+    }
+
     @Test
     public void createCountryToLanguageMapper() throws Exception {
         CountryToLanguageMapper paymentDispatcher = factory.createCountryToLanguageMapper();
         assertThat(paymentDispatcher).isNotNull();
     }
+
+    /**
+     * Because {@link TenantFactory} initializes {@link com.commercetools.pspadapter.payone.PaymentHandler}
+     * and {@link PaymentDispatcher} inside constructor, we can't easily mock an instance, thus we make this anonymous
+     * class extending.
+     *
+     * @param tenantConfig         {@link TenantConfig} to inject
+     * @param blockingSphereClient {@link BlockingSphereClient} to inject
+     * @param payonePostService    {@link PayonePostService} to inject
+     * @return instance of {@link TenantFactory} with injected values from above.
+     */
+    private static TenantFactory mockTenantFactory(TenantConfig tenantConfig, BlockingSphereClient blockingSphereClient, PayonePostService payonePostService) {
+        return new TenantFactory("testPayoneInterfaceName", tenantConfig) {
+            @Nonnull
+            @Override
+            protected BlockingSphereClient createBlockingSphereClient(TenantConfig tenantConfig) {
+                return blockingSphereClient;
+            }
+
+            @Nonnull
+            @Override
+            protected PayonePostService getPayonePostService(TenantConfig tenantConfig) {
+                return payonePostService;
+            }
+        };
+    }
+
+    /**
+     * <table border="1">
+     * <tr><th>CTP</th><td>&nbsp;</td><th>Payone</th></tr>
+     * <tr><td>{@link TransactionType#AUTHORIZATION}</td><td>&nbsp;-&gt;&nbsp;</td><td>{@link RequestType#PREAUTHORIZATION}</td></tr>
+     * <tr><td>{@link TransactionType#CHARGE}</td><td>&nbsp;-&gt;&nbsp;</td><td>{@link RequestType#AUTHORIZATION}</td></tr>
+     * <tr><td>other</td><td>&nbsp;-&gt;&nbsp;</td><td>"unexpected transaction request type "%s""</td></tr>
+     * </table>
```

Comment:
Maybe let's add some human-readable comments too?

Reply:
Not sure what you mean. Javadoc actually are human specific comments:
![image](https://user-images.githubusercontent.com/22023239/36982082-3264b74a-208f-11e8-9766-1405145be3e6.png)